python/src/wrfcloud/runtime/postproc.py

```diff
@@ -20,12 +24,133 @@ def __init__(self, runinfo: RunInfo):
+                          'SpcCoeff/Big_Endian/v.seviri_m10.SpcCoeff.bin',
+                          'TauCoeff/ODPS/Big_Endian/imgr_g11.TauCoeff.bin',
+                          'TauCoeff/ODPS/Big_Endian/imgr_g12.TauCoeff.bin',
+                          'TauCoeff/ODPS/Big_Endian/imgr_g13.TauCoeff.bin',
+                          'TauCoeff/ODPS/Big_Endian/imgr_g15.TauCoeff.bin',
+                          'TauCoeff/ODPS/Big_Endian/imgr_mt1r.TauCoeff.bin',
+                          'TauCoeff/ODPS/Big_Endian/imgr_mt2.TauCoeff.bin',
+                          'TauCoeff/ODPS/Big_Endian/imgr_insat3d.TauCoeff.bin',
+                          'TauCoeff/ODPS/Big_Endian/amsre_aqua.TauCoeff.bin',
+                          'TauCoeff/ODPS/Big_Endian/tmi_trmm.TauCoeff.bin',
+                          'TauCoeff/ODPS/Big_Endian/ssmi_f13.TauCoeff.bin',
+                          'TauCoeff/ODPS/Big_Endian/ssmi_f14.TauCoeff.bin',
+                          'TauCoeff/ODPS/Big_Endian/ssmi_f15.TauCoeff.bin',
+                          'TauCoeff/ODPS/Big_Endian/ssmis_f16.TauCoeff.bin',
+                          'TauCoeff/ODPS/Big_Endian/ssmis_f17.TauCoeff.bin',
+                          'TauCoeff/ODPS/Big_Endian/ssmis_f18.TauCoeff.bin',
+                          'TauCoeff/ODPS/Big_Endian/ssmis_f19.TauCoeff.bin',
+                          'TauCoeff/ODPS/Big_Endian/ssmis_f20.TauCoeff.bin',
+                          'TauCoeff/ODPS/Big_Endian/seviri_m10.TauCoeff.bin' ]
+
+        for fixfile in sat_fix_files:
+            self.symlink(f'{self.runinfo.uppcodedir}/src/lib/crtm2/src/fix/' + fixfile, os.path.basename(fixfile))
+
+    def run_upp(self) -> None:
+        """
+        Executes the unipost.exe program
+        """
+        self.log.debug('Linking unipost.exe to upp working directory')
+        self.symlink(f'{self.runinfo.uppcodedir}/bin/unipost.exe', 'unipost.exe')
+        
+        self.log.debug('Executing unipost.exe')
+        if self.runinfo.wrfcores == 1:
+            upp_cmd = './unipost.exe >& unipost.log'
+            os.system(upp_cmd)
+        else:
+            self.submit_job('unipost.exe',self.runinfo.wrfcores,'wrf')
+
     def run(self) -> bool:
         """
         Main routine that sets up, runs, and monitors post-processing end-to-end
         """
         self.log.info(f'Setting up post-processing for "{self.runinfo.name}"')
-        self.log.warn(f"{__name__} isn't fully implemented yet!")
+
+        # Check if experiment working directory already exists, take action based on value of runinfo.exists
+        action = check_wd_exist(self.runinfo.exists,self.runinfo.uppdir)
+        if action == "skip":
+            return True
+
+        os.mkdir(self.runinfo.uppdir)
+        os.chdir(self.runinfo.uppdir)
+
+        startdate = self.runinfo.startdate
+        startdate = datetime.strptime(startdate, '%Y-%m-%d_%H:%M:%S')
+        enddate   = self.runinfo.enddate
+        enddate   = datetime.strptime(enddate, '%Y-%m-%d_%H:%M:%S')
```

Comment:
My original comment was incorrect, but these lines can be consolidated (similar to logic in `__init__.py`)

```suggestion
        startdate = datetime.strptime(self.runinfo.startdate, '%Y-%m-%d_%H:%M:%S')
        enddate   = datetime.strptime(self.runinfo.enddate, '%Y-%m-%d_%H:%M:%S')
```

Reply:
This didn't work originally but I can't remember why, it may have been because i was originally only doing an import datetime. I will try to this again and test